pkg/services/dashboards/accesscontrol.go

```diff
@@ -14,6 +20,26 @@ const (
 )
 
 var (
-	ScopeFoldersAll      = accesscontrol.GetResourceAllScope(ScopeFoldersRoot)
-	ScopeFoldersProvider = accesscontrol.NewScopeProvider(ScopeFoldersRoot)
+	ScopeFoldersAll      = ac.GetResourceAllScope(ScopeFoldersRoot)
+	ScopeFoldersProvider = ac.NewScopeProvider(ScopeFoldersRoot)
 )
+
+// NewNameScopeResolver provides an AttributeScopeResolver that is able to convert a scope prefixed with "folders:name:" into an id based scope.
+func NewNameScopeResolver(db Store) (string, ac.AttributeScopeResolveFunc) {
+	prefix := ScopeFoldersProvider.GetResourceScopeName("")
+	resolver := func(ctx context.Context, orgID int64, scope string) (string, error) {
+		if !strings.HasPrefix(scope, prefix) {
+			return "", ac.ErrInvalidScope
+		}
+		nsName := scope[len(prefix):]
+		if nsName == "*" {
```

Comment:
Is `*` a valid dashboard/folder name? 🤔
Could that be abused?

Reply:
I just tested that, and we can name a folder `*`. So this would mean that we'd need permissions scoped with `GetResourceAllIDScope` to access such a folder.
This is the same for the scope resolver we made for data sources. We need to change it there as well.

Reply:
Hmm, this might be another benefit of @kalleep's [suggestion to remove the prefix for wildcard scopes](https://docs.google.com/document/d/1JHxe_k4OhI2QqgTZMU2qkxpWF9Dgjfvw-PvyF6IJQIs/edit#) 🤔 

This also seems like a pretty rare scenario, so I think it might be ok to merge this as it is and rethink how we deal with this later? But perhaps I'm underestimating this.

Reply:
I think it makes sense to remove this shortcut at all. I was mistaken by an implementation for datasources.